tests/python/rest_api/test_webhooks.py

```diff
@@ -0,0 +1,1045 @@
+            for o in organizations
+            for u in find_users(org=o['id'], role=role, exclude_privilege='admin')
+            for p in projects_by_org.get(o['id'], [])
+            if p['owner']['id'] == u['id']
+        ))
+
+        webhook = deepcopy(self.proj_webhook)
+        webhook['project_id'] = pid
+
+        response = post_method(username, 'webhooks', webhook, org_id=oid)
+
+        assert response.status_code == HTTPStatus.CREATED
+        assert 'secret' not in response.json()
+
+    def test_non_member_cannot_create_webhook_for_org(self, find_users, organizations,
+            is_org_member):
+        username, org_id = next((
+            (u['username'], o['id'])
+            for o in organizations
+            for u in find_users(exclude_privilege='admin')
+            if not is_org_member(u['id'], o['id'])
+        ))
+
+        webhook = deepcopy(self.org_webhook)
+
+        response = post_method(username, 'webhooks', webhook, org_id=org_id)
+
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+
+    def test_can_create_without_unnecessary_fields(self):
+        post_data = deepcopy(self.proj_webhook)
+        post_data.pop('enable_ssl')
+        post_data.pop('content_type')
+        post_data.pop('description')
+        post_data.pop('is_active')
+        post_data.pop('secret')
+
+        response = post_method('admin2', 'webhooks', post_data)
+
+        assert response.status_code == HTTPStatus.CREATED
+
+    def test_cannot_create_without_target_url(self):
+        post_data = deepcopy(self.proj_webhook)
+        post_data.pop('target_url')
+
+        response = post_method('admin2', 'webhooks', post_data)
+
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+
+
+    def test_cannot_create_without_events_list(self):
+        post_data = deepcopy(self.proj_webhook)
+        post_data.pop('events')
+
+        response = post_method('admin2', 'webhooks', post_data)
+
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+
+    def test_cannot_create_without_type(self):
+        post_data = deepcopy(self.proj_webhook)
+        post_data.pop('type')
+
+        response = post_method('admin2', 'webhooks', post_data)
+
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+
+    def test_cannot_create_without_project_id(self):
+        post_data = deepcopy(self.proj_webhook)
+        post_data.pop('project_id')
+
+        response = post_method('admin2', 'webhooks', post_data)
+
+        assert response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR
+
+    def test_cannot_create_organization_webhook_when_project_id_is_not_null(self, organizations):
+        post_data = deepcopy(self.proj_webhook)
+        post_data['type'] = 'organization'
+        org_id = organizations.raw[0]['id']
+
+        response = post_method('admin2', 'webhooks', post_data, org_id=org_id)
+
+        assert response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR
```

Comment:
@kirill-sizov , if a user provides wrong arguments, server has to reply with 400 error. We should not return INTERNAL_SERVER_ERROR. It is a problem with the server itself. Even if our current server can return, it is a bug which should be fixed in the future.